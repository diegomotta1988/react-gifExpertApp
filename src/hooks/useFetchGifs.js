import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const API_KEY = 'jK8GqN6qzwSVhjjH1UwKaRNVqJijd5Ig';
  const [state, setState] = useState({ images: [], loading: true });

  useEffect(() => {
    getGifs(category, API_KEY).then((images) => {
      setState({ images, loading: false });
    });
  }, [category]);

  return state;
};
