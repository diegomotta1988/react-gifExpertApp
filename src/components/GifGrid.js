import React from 'react';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from './../hooks/useFetchGifs';

import propTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const { data: images, loading } = useFetchGifs(category);
  // Al pasarle las dependencias como [] s√≥lo se ejecuta al renderizarse por primera vez el componente

  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{category}</h3>
      {loading && <p className='animate__animated animate__flash'>Loading</p>}
      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: propTypes.string.isRequired,
};
