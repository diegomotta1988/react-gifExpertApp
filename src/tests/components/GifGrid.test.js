import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from './../../hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas en el <GifGrid />', () => {
  const category = 'Dragon Ball';
  const mockUseFetchGifs = useFetchGifs;

  test('debe mostrarse correctamente', () => {
    mockUseFetchGifs.mockImplementation((category) => ({
      data: [],
      loading: true,
    }));
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar items cuando se cargan imágenes con useFecthGifs', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://test',
        title: 'test',
      },
    ];
    mockUseFetchGifs.mockImplementation((category) => ({
      data: gifs,
      loading: false,
    }));
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    // Comprobamos que no hay párrafo de cargando
    expect(wrapper.find('p').exists()).toBe(false);
    // Comprobamos que genera un item por cada gif
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
