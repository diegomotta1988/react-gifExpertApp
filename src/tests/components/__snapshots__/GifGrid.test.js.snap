// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <GifGrid /> debe mostrar items cuando se cargan im√°genes con useFecthGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="test"
      url="https://test"
    />
  </div>
</Fragment>
`;

exports[`pruebas en el <GifGrid /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
